pkg/controller/endpointslicemirroring/utils.go

```diff
@@ -88,10 +88,7 @@ func newEndpointSlice(endpoints *corev1.Endpoints, ports []discovery.EndpointPor
 	ownerRef := metav1.NewControllerRef(endpoints, gvk)
 	epSlice := &discovery.EndpointSlice{
 		ObjectMeta: metav1.ObjectMeta{
-			Labels: map[string]string{
-				discovery.LabelServiceName: endpoints.Name,
-				discovery.LabelManagedBy:   controllerName,
-			},
+			Labels:          endpoints.Labels,
```

Comment:
this needs to copy the endpoints.Labels map to avoid mutating the informer-owned labels map when we assign LabelServiceName/LabelManagedBy below

Reply:
add a test that verifies the endpoints object passed into `newEndpointSlice` does not have its labels modified

Reply:
Thanks for catching this! I've updated this to copy labels and added a test to verify that the labels on the Endpoints resource are not mutated.